govcd/adminorg.go

```diff
@@ -362,23 +363,58 @@ func (adminOrg *AdminOrg) removeAllOrgNetworks() error {
 
 // Forced removal of all organization catalogs
 func (adminOrg *AdminOrg) removeCatalogs() error {
-	for _, catalogs := range adminOrg.AdminOrg.Catalogs.Catalog {
-		// Get Catalog HREF
-		catalogHREF := adminOrg.client.VCDHREF
-		catalogHREF.Path += "/admin/catalog/" + strings.Split(catalogs.HREF, "/api/admin/catalog/")[1] //gets id
-		req := adminOrg.client.NewRequest(map[string]string{
-			"force":     "true",
-			"recursive": "true",
-		}, http.MethodDelete, catalogHREF, nil)
-		_, err := checkResp(adminOrg.client.Http.Do(req))
+	for _, catalog := range adminOrg.AdminOrg.Catalogs.Catalog {
+		isCatalogFromSameOrg, err := isCatalogFromSameOrg(adminOrg, catalog.Name)
 		if err != nil {
-			return fmt.Errorf("error deleting catalog: %s, %s", err, catalogHREF.Path)
+			return fmt.Errorf("error deleting catalog: %s", err)
+		}
+		if isCatalogFromSameOrg {
+			// Get Catalog HREF
+			catalogHREF := adminOrg.client.VCDHREF
+			catalogHREF.Path += "/admin/catalog/" + strings.Split(catalog.HREF, "/api/admin/catalog/")[1] //gets id
+			req := adminOrg.client.NewRequest(map[string]string{
+				"force":     "true",
+				"recursive": "true",
+			}, http.MethodDelete, catalogHREF, nil)
+			_, err := checkResp(adminOrg.client.Http.Do(req))
+			if err != nil {
+				return fmt.Errorf("error deleting catalog: %s, %s", err, catalogHREF.Path)
+			}
 		}
 	}
 	return nil
 
 }
 
+// isCatalogFromSameOrg checks if catalog is in same Org. Shared catalogs form other Org are showed as normal one
+// in some API responses.
+func isCatalogFromSameOrg(adminOrg *AdminOrg, catalogName string) (bool, error) {
+	foundCatalogs, err := adminOrg.FinAdminCatalogByName(catalogName, adminOrg.AdminOrg.Name)
+	if err != nil {
+		return false, err
+	}
+
+	if len(foundCatalogs) == 1 {
+		return true, nil
+	}
+	return false, nil
+}
+
+// FinCatalogByName uses catalog name and Org name to return CatalogRecordType information.
+func (adminOrg *AdminOrg) FinAdminCatalogByName(name, orgName string) ([]*types.CatalogRecordType, error) {
```

Comment:
Three problems:
1. The method name is misspelled. It should be `Find`, not `Fin`.
2. According to our conventions, the method should indicate what object is retrieving. Not `FindAdminCatalog` but`FindCatalogRecordTypes` (I believe we can skip the `ByName` part)
3. Why do we pass orgName? Isn't the org name already in the receiver `adminOrg` data?

Reply:
fixed